core/src/main/scala/pureconfig/DurationConvert.scala

```diff
@@ -11,13 +15,8 @@ private[pureconfig] object DurationConvert {
   /**
    * Convert a string to a Duration while trying to maintain compatibility with Typesafe's abbreviations.
    */
-  def fromString[D](durationString: String, ct: ClassTag[D]): Try[Duration] = {
-    Try { Duration(addZeroUnit(justAMinute(itsGreekToMe(durationString)))) }
-      .recoverWith {
-        case ex: NumberFormatException =>
-          val err = s"Could not parse a ${ct.runtimeClass.getSimpleName} from '$durationString'. (try ns, us, ms, s, m, h, d)"
-          new util.Failure(new IllegalArgumentException(err, ex))
-      }
+  val fromString: String => Either[ConfigReaderFailure, Duration] = {
+    catchReadError(string => Duration(addZeroUnit(justAMinute(itsGreekToMe(string)))))
```

Comment:
I've improved the error message with 005753f. Note that I've decided to follow another route because the error message can be found inside the exception and the string is already in the `CannotConvert` failure. Basically we don't use `Try` or throw exceptions anymore and that should improve everything :smile: